packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -629,13 +629,35 @@ def __repr__(self) -> str:
         """Return a string representation of the data."""
         return f"AbciAppDB({self._data})"
 
-    def cleanup(self, cleanup_history_depth: int) -> None:
-        """Reset the db."""
+    def cleanup(
+        self,
+        cleanup_history_depth: int,
+        cleanup_history_depth_current: Optional[int] = None,
+    ) -> None:
+        """Reset the db, keeping only the latest entries (periods).
+
+        If cleanup_history_depth_current has been also set, also clear oldest historic values in the current entry.
+        """
         cleanup_history_depth = max(cleanup_history_depth, MIN_HISTORY_DEPTH)
         self._data = {
             key: self._data[key]
             for key in sorted(self._data.keys())[-cleanup_history_depth:]
         }
+        self.cleanup_current_histories(cleanup_history_depth_current)
+
+    def cleanup_current_histories(
+        self, cleanup_history_depth_current: Optional[int]
+    ) -> None:
+        """Reset the parameter histories for the current entry (period), keeping only the latest values for each parameter."""
+        if not cleanup_history_depth_current:
+            return
```

Comment:
would be cleaner to read if it was above on 646 and the method just does the cleaning